### PR TITLE
Update PSR-4-autoloader-examples.md

### PR DESCRIPTION
**Pull request by WinterSilence**
### Please, don't close - this PR using in Github bug report, I'm will close him, when ticket done.

Fix directory separator, long/strange conditions. 
Optimize method `loadMappedFile`: move up `str_replace`.

### CODE REVIEW

**Comment by WinterSilence** [comment]
Also, need check  `.php` in result `spl_autoload_extensions()`, updated method `register`:
~~~
public function register($prepend = false)
{
    \spl_autoload_register(array($this, 'loadClass'), true, (bool) $prepend);

    $extensions = \explode(',', \spl_autoload_extensions());
    if ($extensions[0] != '.php') {
        \array_unshift($extensions, '.php');
        \spl_autoload_extensions(\implode(',', \array_unique($extensions)));
    }
}
~~~

File: accepted/PSR-4-autoloader-examples.md [hunk 1]
```diff
@@ -166,11 +166,11 @@ class Psr4AutoloaderClass
     public function loadClass($class)
     {
         // the current namespace prefix
-        $prefix = $class;
+        $prefix = $class = ltrim($class, '\\');
 
         // work backwards through the namespace names of the fully-qualified
         // class name to find a mapped file name
-        while (false !== $pos = strrpos($prefix, '\\')) {
```
Comment (goetas) [inline accepted/PSR-4-autoloader-examples.md]: Is this change correct? What if the chat is at position 0?

Comment (WinterSilence) [reply accepted/PSR-4-autoloader-examples.md]: @samdark @goetas `$prefix = $class = ltrim($class, '\\');`

File: accepted/PSR-4-autoloader-examples.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ spl_autoload_register(function ($class) {
     $prefix = 'Foo\\Bar\\';
 
     // base directory for the namespace prefix
-    $base_dir = __DIR__ . '/src/';
+    $base_dir = __DIR__ . DIRECTORY_SEPARATOR . 'src' . DIRECTORY_SEPARATOR;
```
[the remark this answers is not in the document]
Comment (WinterSilence) [reply accepted/PSR-4-autoloader-examples.md]: @samdark it's need for realpath in php < 5.6

File: accepted/PSR-4-autoloader-examples.md (hunk 1 above)
Comment (samdark) [reply accepted/PSR-4-autoloader-examples.md]: OK. This one would work.

**Comment by WinterSilence** [comment]
@samdark Don't see reason to change `DIRECTORY_SEPARATOR`: it's granted by PHP team and working **everywhere**. I just follow their recommendations. If you don't agree, discuss with PHP devs.

**Comment by WinterSilence** [comment]
@samdark 
> not checking for false in strpos

no reason because `$prefix = $class = ltrim($class, '\\');`

**Pull request closed without merging**